pkg/v1/mutate/mutate.go

```diff
@@ -392,3 +421,85 @@ func inWhiteoutDir(fileMap map[string]bool, file string) bool {
 	}
 	return false
 }
+
+//CREATED is a constant timestamp, set to the 0 time.
+var CREATED = time.Time{}
+
+// StripConfig strips out any timestamp from the layers and config of an image.
+func StripConfig(img v1.Image) (v1.Image, error) {
+	newImage := empty.Image
+
+	layers, err := img.Layers()
+	if err != nil {
+		return nil, err
+	}
+
+	// Strip away all timestamps from layers
+	var newLayers []v1.Layer
+	for _, layer := range layers {
+		newLayer, err := stripLayer(layer)
```

Comment:
I think calling `stripLayer` here makes `StripConfig` a misleading name.

Reply:
I agree that `StripConfig` is a bad name, I had it there more as a placeholder. Do you have an idea for a better name? `StripTimestamps`?

Reply:
`StripTimestamps` is better, but this is also an opportunity to coin a new humorous term that is more fun to say, e.g. `Reproducibilify`, `TimeTravel`, or `Warp`.

Reply:
The library function name should probably be boring, but I'm +1 on giving it a fun name in crane or some other CLI.

How about stripmetadata for now?

Reply:
It's not necessarily "stripping" the metadata/timestamps.

This reads in a nice way, I think:
```go
frozen, err := mutate.Time(img, time.Now())
```

Reply:
I like that - but this actually does a bit more than remove times (there are some random numbers in there too we remove). Any thoughts on that?

Reply:
I think I'd prefer two methods, then. One to set the time and one to remove various randomness. I would expect clients to compose these to achieve what they want. If I'm reading this correctly, the random numbers are just in the config file, so there shouldn't be any performance issues with splitting it up into two mutations:

```go
frozen, _ := mutate.Time(img, whenever)
happy, _ := mutate.ConfigFile(img, notRandom)
```

If we find a need to use this all over the place, we could have a helper do both of those, maybe:
```go
canon, _ := mutate.Canonicalize(img)
```

Which strips all the metadata and sets the time to something we hard-code. WDYT?

Reply:
Makes sense, I'll make the switch.